### ISSUE TITLE
Speed up library dependency indexing

### ISSUE THREAD

**Issue by olafurpg** [scalameta/metals#535]
Opening the bloop repo with `-Dmetals.statistics=index`

```
time: connected to build server in 0.42s
time: imported build in 3.86s
time: updated build targets in 0.12s
time: started file watcher in 0.58s
time: indexed library classpath in 2.1s
time: indexed workspace SemanticDBs in 0.45s
time: indexed workspace sources in 0.49s
time: indexed library sources in 6.17s <-- bottleneck
time: indexed workspace in 9.91s
```

The main bottleneck is "indexed library sources", which builds the following index
```
memory: definition index using 23.2M (1,772,885 lines Scala)
```
See https://scalameta.org/metals/blog/2019/01/22/bloom-filters.html#indexing-time for a flamegraph.

We should be able to persist that index so we only process `*-sources.jar` files once. That would speed subsequent startups for a lot of codebases, especially ones that have dependencies with large number of sources.

**Comment by olafurpg** [comment]
Open PR #534 by @mudsam implementing this feature 🎉

**Comment by olafurpg** [comment]
Fixed in #534
